packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -196,3 +189,15 @@ export const WithHintItemHint = styled.span`
 	padding-inline-end: ${ space( 1 ) };
 	margin-block: ${ space( 1 ) };
 `;
+
+export const SelectedItemCheck = styled( Ariakit.SelectItemCheck )`
+	display: flex;
+	align-items: center;
+	margin-inline-start: ${ space( 2 ) };
+	font-size: 0;
+
+	${ WithHintItemWrapper } ~ &,
+	[aria-selected='true'] & {
```

Comment:
> what do you think about this proposed change?

I guess there are many ways to address this problem, but I think this approach is ok for now as long as we add a code comment. Otherwise it isn't immediately clear what this font-size switching is for.

There's a part of me that want to address it at the JS level and not in CSS, but given that the checkmark might be moved to left eventually, it's probably not worth the effort to rejigger the code now.

Also, no strong opinion but I think I slightly prefer this since it captures how `SelectedItemCheck` is some kind of placeholder:

```suggestion
	&:not(:empty) {
```

(Though, the change may not be necessary as long as the code comment is explaining things)

Reply:
> There's a part of me that want to address it at the JS level and not in CSS, but given that the checkmark might be moved to left eventually, it's probably not worth the effort to rejigger the code now.

I had the same itch, but I figured that seeing how V2 may diverge from V1 and its quirks, we may just end up leaving future V1 implementation independent (at least the styles).

I'll go ahead and implement your suggestions before merging.

Reply:
Changes extracted in #63229. Once merged, I will rebase and merge this PR too.